### PR TITLE
OBPIH-4769 Improvements to the products list API

### PR DESCRIPTION
**Pull request by awalkowiak**

### CODE REVIEW

File: grails-app/services/org/pih/warehouse/product/ProductService.groovy [hunk 1]
```diff
@@ -271,7 +271,9 @@ class ProductService {
      * @return
      */
     List<Product> getProducts(Category category, List<Tag> tags, boolean includeInactive, Map params) {
-        return getProducts(category, [], tags, includeInactive, params)
+        def categories = []
+        categories << category
```
Comment (jmiranda) [inline grails-app/services/org/pih/warehouse/product/ProductService.groovy:275]: I think you can shorten this to 
```
def categories = [category]
```

**Review by jmiranda — COMMENTED** [review]
LGTM

File: grails-app/services/org/pih/warehouse/product/ProductService.groovy (hunk 1 above)
Comment (awalkowiak) [reply grails-app/services/org/pih/warehouse/product/ProductService.groovy:275]: Yup, you're right
